patches/additionalPythonPackages/default.nix

```diff
@@ -49,6 +49,12 @@ in
 
                             in  allPythonRecDep;
 
+  # function to fetch source tarball from BBP devpi server, given package name / version / sha256
+  fetchBBPDevpi = { pname, version, sha256 }:
+    let
+      url = "https://bbpteam.epfl.ch/repository/devpi/bbprelman/release/+f/${builtins.substring 0 3 sha256}/${builtins.substring 3 13 sha256}/${pname}-${version}.tar.gz";
```

Comment:
unfortunately, sha256 makes a part of the only direct URL to source tarball that I have found